open-apparel-registry/open-apparel-registry
# Enable viewing and editing profiles

## Overview

This PR enables viewing and editing profile data.

For logged in users, their profiles are now editable as depicted below. Edits required a password for confirmation and are limited to the following fields:

- name
- description
- contributor type
- other contributor type
- website

... all of which live on the contributor model

Profiles for other users are view-only.

I also updated the Facility Details page to have links to the contributor profiles.

Connects #104

## Demo

![view-edit-profile](https://user-images.githubusercontent.com/4165523/53846418-d2667200-3f7a-11e9-867c-f7b23e2aff89.gif)
![facility-profile-link](https://user-images.githubusercontent.com/4165523/53846419-d2667200-3f7a-11e9-9680-bef92e9bc7ac.gif)
![edit-profile-errors](https://user-images.githubusercontent.com/4165523/53846420-d2667200-3f7a-11e9-88bd-61f1958dfede.gif)

## Notes

0f6e4cd  is an unrelated commit to fix a map loading bug, but I have it here so I could test it out in the course of working on this.

## Testing

- get this branch, then resetdb and register for two accounts
- sign in to one of the accounts and visit your profile page, verifying that it shows up as editable
- search for some facilities and then visit a details page. verify that there is now a link from any contributors to their profile pages and that those are view-only
- edit your profile and attempt to save it. verify that
   - not including a required field leads to an error message
   - an incorrect password shows an error message
   - successfully saving shows the toast and updates your profile data and, if it's different, the name in the app header
- run `./scripts/test` and verify that they pass
